util.go

```diff
@@ -46,7 +47,7 @@ func readString(r io.Reader, n int) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	return string(b), nil
+	return string(bytes.Trim(b, "\x00")), nil
```

Comment:
This is used in a lot of places.  Double check that your test example is correctly setting the length of the string value (n) which is used to work out how many bytes to read for the string.

Reply:
I agree we need to be carefull when touching this part of the code, but either we implement this in the utils, or we need to add it in the id3v11 parser. This (I think) only related to the id3v11 tags, because the tag component have a constent size (30 for artist/title/album).

https://github.com/dhowden/tag/blob/master/id3v1.go#L62

Reply:
Do you think I should move this `bytes.Trim(b, "\x00")` function into the id3v1 file ?

Reply:
Yes, this is a better idea.

For id3v1 the fixed-width fields are either space or zero padded, we should trim both in the same place: https://github.com/dhowden/tag/blob/master/id3v1.go#L101